dll/win32/kernel32/client/sysinfo.c

```diff
@@ -446,8 +636,81 @@ GetSystemFirmwareTable(IN DWORD FirmwareTableProviderSignature,
                        OUT PVOID pFirmwareTableBuffer,
                        IN DWORD BufferSize)
 {
-    STUB;
-    return 0;
+    /* This function currently obtains data using a hack (registry workaround)
+       which is located here: drivers/input/i8042prt/hwhacks.c
+       i8042StoreSMBiosTables is responsible for writing SMBIOS table into registry
+
+       Should be implemented correctly using NtQuerySystemInformation
+       along with SystemFirmwareTableInformation class
+
+       Reference: https://github.com/hfiref0x/VMDE/blob/master/src/vmde/sup.c
+     */
+
+    UINT uSize = 0;
+    DWORD dwError = ERROR_SUCCESS;
+    PVOID pBuffer = NULL;
+
+    switch (FirmwareTableProviderSignature)
+    {
+        case 'ACPI':
+        {
+            /* FIXME: Not implemented yet */
+            dwError = ERROR_NOT_FOUND;
+            break;
+        }
+        case 'FIRM':
+        {
+            /* FIXME: Not implemented yet */
+            dwError = ERROR_INVALID_PARAMETER;
```

Comment:
Why not having consistent (& relevant) error codes?

Reply:
This is what Windows returns, when the table with specified identifier is not found.

But ok, returning NOT_IMPLEMENTED would be more correct. Fixed.